R/create_service_use_lookup.R

```diff
@@ -0,0 +1,776 @@
+      geriatric_cohort = assign_geriatric_cohort(.data$geriatric_cost),
+      elective_inpatient_cohort = assign_elective_inpatient_cohort(.data$elective_inpatient_flag),
+      limited_daycases_cohort = assign_limited_daycases_cohort(.data$elective_inpatient_flag, .data$elective_instances),
+      routine_daycase_cohort = assign_routine_daycase_cohort(.data$elective_inpatient_flag, .data$elective_instances),
+      single_emergency_cohort = assign_single_emergency_cohort(.data$emergency_instances),
+      multiple_emergency_cohort = assign_multiple_emergency_cohort(.data$emergency_instances),
+      prescribing_cohort = assign_prescribing_cohort(.data$prescribing_cost),
+      outpatient_cohort = assign_outpatient_cohort(.data$outpatient_cost),
+      ae2_cohort = assign_ae2_cohort(.data$ae2_cost),
+      community_care_cohort = assign_community_care_cohort(.data$community_health_cost),
+      # Assign other cohort if none have been assigned
+      other_cohort = rowSums(dplyr::across("psychiatry_cohort":"community_care_cohort")) == 0,
+
+      # Recalculate costs based on the cohorts
+      elective_inpatient_cost = recalculate_elective_inpatient_cost(
+        .data$elective_inpatient_cohort,
+        .data$acute_elective_cost
+      ),
+      limited_daycases_cost = calculate_limited_daycases_cost(
+        .data$limited_daycases_cohort,
+        .data$acute_elective_cost
+      ),
+      routine_daycase_cost = calculate_routine_daycase_cost(
+        .data$routine_daycase_cohort,
+        .data$acute_elective_cost
+      ),
+      single_emergency_cost = calculate_single_emergency_cost(
+        .data$single_emergency_cohort,
+        .data$acute_emergency_cost
+      ),
+      multiple_emergency_cost = calculate_multiple_emergency_cost(
+        .data$multiple_emergency_cohort,
+        .data$acute_emergency_cost
+      ),
+      community_care_cost = calculate_community_care_cost(
+        .data$community_care_cohort,
+        .data$community_health_cost
+      ),
+      # Care Home cost is removed for now, so set to zero
+      residential_care_cost = calculate_residential_care_cost(),
+      # Replace any missing total costs with zero
+      dplyr::across("cost_total_net", ~ replace(., is.na(.), 0))
+    ) %>%
+    # Add the cohort names
+    assign_cohort_names() %>%
+    # Select out the required variables
+    dplyr::select(
+      "chi",
+      "service_use_cohort",
+      "psychiatry_cost",
+      "maternity_cost",
+      "geriatric_cost",
+      "elective_inpatient_cost",
+      "limited_daycases_cost",
+      "single_emergency_cost",
+      "multiple_emergency_cost",
+      "routine_daycase_cost",
+      "outpatient_cost",
+      "prescribing_cost",
+      "ae2_cost"
+    )
+
+  if (write_to_disk == TRUE) {
+    write_rds(return_data,
+      path = glue::glue("{get_slf_dir()}/Cohorts/Service_Use_Cohorts_{year}.rds")
+    )
+  } else {
+    return(return_data)
+  }
```

Comment:
can use the function `get_service_use_cohorts_path` here

Reply:
```suggestion
  if (write_to_disk == TRUE) {
    write_rds(return_data,
      path = get_service_use_cohorts_path(year)
    )
  } 
  
return(return_data)
```